integration-tests/tests/src/utils/import-app.ts

```diff
@@ -33,52 +34,142 @@ function getUrls() {
   };
 }
 
+function getIsAppImporterRemote() {
+  const { isAppImporterRemote } = environment;
+  return isAppImporterRemote === "true";
+}
+
+type Credentials =
+  | {
+      kind: "api-key";
+      publicKey: string;
+      privateKey: string;
+    }
+  | {
+      kind: "username-password";
+      username: string;
+      password: string;
+    };
+
+function getCredentials(): Credentials {
+  const { publicKey, privateKey, username, password } = environment;
+  if (typeof publicKey === "string" && typeof privateKey === "string") {
+    return {
+      kind: "api-key",
+      publicKey,
+      privateKey,
+    };
+  }
+  return {
+    kind: "username-password",
+    username: typeof username === "string" ? username : "unique_user@domain.com",
+    password: typeof password === "string" ? password : "password",
+  };
+}
+
 export function getDefaultReplacements(name: string): TemplateReplacements {
   // When running on CI we connect through mongodb-atlas instead of local-mongodb
   const { mongodbClusterName } = environment;
-  if ((name === "with-db" || name === "with-db-flx") && typeof mongodbClusterName === "string") {
-    return {
-      "services/mongodb/config.json": {
-        type: "mongodb-atlas",
-        config: {
-          clusterName: mongodbClusterName,
-          readPreference: "primary",
-          wireProtocolEnabled: false,
+  if (typeof mongodbClusterName === "string") {
+    if (name === "with-db") {
+      return {
+        "services/mongodb/config.json": {
+          type: "mongodb-atlas",
+          config: {
+            clusterName: mongodbClusterName,
+            readPreference: "primary",
+            wireProtocolEnabled: false,
+            sync: {
+              database_name: `test-database-${new BSON.ObjectID().toHexString()}`,
+            },
+          },
         },
-      },
-    };
-  } else {
-    return {};
+      };
+    } else if (name === "with-db-flx") {
+      return {
+        "services/mongodb/config.json": {
+          type: "mongodb-atlas",
+          config: {
+            clusterName: mongodbClusterName,
+            readPreference: "primary",
+            wireProtocolEnabled: false,
+            flexible_sync: {
+              database_name: `test-database-${new BSON.ObjectID().toHexString()}`,
+            },
+          },
+        },
+      };
+    }
```

Comment:
This could probably be combined and setup with inline ternary determined properties